### PR TITLE
Trace more SASL calls

### PR DESCRIPTION
**Pull request by tiran**
Add log trace for sasl_server_start() and sasl_server_step() to
investigate performance issues with SASL bind.

See: https://github.com/389ds/389-ds-base/issues/4506
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ldap/servers/slapd/saslbind.c [hunk 1]
```diff
@@ -945,9 +993,7 @@ ids_sasl_check_bind(Slapi_PBlock *pb)
             goto sasl_start;
         }
 
-        rc = sasl_server_step(sasl_conn,
-                              cred->bv_val, cred->bv_len,
-                              &sdata, &slen);
+        rc = ids_sasl_server_step(pb_conn, cred, &sdata, &slen);
```
Comment (Firstyear) [inline ldap/servers/slapd/saslbind.c:996]: Depending on what we want to do here, we could add these in #ifdef DEBUG.

**Comment by mreynolds389** [comment]
This is somewhat off topic and directed at the Dev team, but we have a problem with Trace Function Call logging level.  It basically makes the server unusable.  The server start up will fail because it times in when this logging level is enabled, and same for other processes.  It's been over-used in the source, and I do not what to use that logging level in this PR.  

@tiran maybe in this case connection logging is the next best choice (SLAPI_LOG_CONNS).

For Dev, we either need new logging levels or we need to change/remove some of the current "trace function call" logging statements.  They need to be reduced, one way or another...

**Comment by Firstyear** [comment]
@mreynolds389 Well, we need to do a lot related to logging in general, but this is a good point for certain.

**Comment by mreynolds389** [comment]
@tiran if you change the log level from SLAPI_LOG_TRACE to SLAPI_LOG_CONNS you get my ack.  

@Firstyear I don't think we should #ifdef this.  This logging could be useful in customer situations, so it should be available with the standard build.

File: ldap/servers/slapd/saslbind.c [hunk 2]
```diff
@@ -751,6 +751,54 @@ ids_sasl_server_new(Connection *conn)
...
+int
+ids_sasl_server_start(Connection *conn, const char *mech,
+                      struct berval *cred,
+                      const char **sdata, unsigned int *slen)
+{
+    int rc;
+    sasl_conn_t *sasl_conn;
+
+    slapi_log_err(SLAPI_LOG_TRACE, "ids_sasl_server_start", "=> (mech=%s)\n",
+                  mech);
```
Comment (mreynolds389) [inline ldap/servers/slapd/saslbind.c:766]: I think we should also log the connection ID in the logging (before and after):

    slapi_log_err(SLAPI_LOG_CONNS, "ids_sasl_server_start", "=> (conn=%" PRIu64 " mech=%s)\n",
                  conn->c_connid, mech);
    ...
    slapi_log_err(SLAPI_LOG_CONNS, "ids_sasl_server_start", "<= (conn=%" PRIu64 " rc=%s)\n",
                  conn->c_connid, sasl_errstring(rc, NULL, NULL));

**Comment by mreynolds389** [comment]
I took this PR and created a new one with more logging improvements:  https://github.com/389ds/389-ds-base/pull/4749

I'm going to close this PR in favor of the new one.  Thanks @tiran !

**Pull request closed without merging**